### DISCUSSION TITLE
Question: Typing Indicator for Bot?

### DISCUSSION THREAD

**Discussion by cfelbertp** [chatwoot/chatwoot#7234] [Q&A]
I have configured a bot for an inbox. However, the response times for the bot are sometimes a bit longer. Therefore, it would be nice if I could show and hide a typing indicator via the API. Is it correct that this is currently not possible? Does anyone perhaps have another idea for an implementation.

Currently, I do it like this: After the user input, I create a message with the characters '...' and when the bot's answer is ready, I delete the message again and then send the actual message. In my opinion, this looks a bit strange for the user.

**Comment by pranavrajs** [comment]
@cfelbertp There are no APIs for this, we can consider adding an API for the same

**Comment by juliooidella** [comment]
Hello! I discovered that it is possible to send the typing event to the user using the API.
Here's an example of how I did it:

post: your_IP://api/v1/accounts/{your_account_ID}/conversations/{id_conversation}/toggle_typing_status

JSON:
{

   "typing_status": "on",
   "is_private": false

}

Header: api_access_token: {user_token}

Tip: typing_status use on to activate and off to deactivate

**Comment by sojan-official** [comment]
tracking in https://github.com/chatwoot/chatwoot/issues/8928
